### ISSUE TITLE
Runtime scope in transitive dependencies

### ISSUE THREAD

**Issue by sdeleuze** [Ninja-Squad/springmockk#21]
That's seems similar to #14. When I try to upgrade https://github.com/spring-guides/tut-spring-boot-kotlin to Gradle 5.3.1, compilation fails with this message:

```
e: /home/seb/workspace/tut-spring-boot-kotlin/src/test/kotlin/com/example/blog/HttpControllersTests.kt: (4, 8): Unresolved reference: io
e: /home/seb/workspace/tut-spring-boot-kotlin/src/test/kotlin/com/example/blog/HttpControllersTests.kt: (27, 3): Unresolved reference: every
e: /home/seb/workspace/tut-spring-boot-kotlin/src/test/kotlin/com/example/blog/HttpControllersTests.kt: (41, 3): Unresolved reference: every
> Task :compileTestKotlin FAILED
```

I have to add explicitly the `implementation("io.mockk:mockk:1.9.1")` dependency. I am not sure `runtime` is the right scope for SpringMockK transitive dependencies like Mockk. Maybe its Gradle configuration should be `api` in order to get the right scope in the published artifiact metadata?

**Comment by jnizet** [comment]
I thought that anyone would first chose to add MockK to their dependencies, and then maybe add SpringMockK. 
And the public API of SpringMockK (i.e. the annotations) is actually not dependent on MockK, so strictly speaking, you don't need MockK in the compile dependencies to use SpringMockK.

But you're the second one to complain, and it would indeed be quite strange to use SpringMockK without ever using MockK in the test classes. So I'll change the scope of MockK

I'm not sure if I should do the same for kotlin-stdlib-jdk8 and spring-boot-starter-test too, though. These should really be added explicitly to the project's dependencies, IMO, and not be added transitively by SpringMockK, so I'll start by leaving them as implementation (runtime in Maven lingo) dependencies.

I was sloppy in defining my dependencies: I depend on the spring boot test starter, although I actually don't need all the dependencies fetched by the starter (and I even need to exclude some). So I'll also sanitize that.

**Comment by sdeleuze** [comment]
Yeah make sense, just changing the scope for MockK should be fine.

**Comment by jnizet** [comment]
I just published version 1.1.2. It's available in jcenter, and should be available in Maven central soon.

**Comment by sdeleuze** [comment]
Thanks!
